lib/Dialect/HW/HWTypes.cpp

```diff
@@ -392,6 +392,8 @@ std::optional<size_t> EnumType::indexOf(mlir::StringRef field) {
   return {};
 }
 
+size_t EnumType::getBitWidth() { return llvm::Log2_64_Ceil(getFields().size()); }
```

Comment:
I think you (unfortunately) have to introduce yet another 1-check in case that the enum has only one field.
> return std::min(1, llvm::Log2_64_Ceil(getFields().size()));

Reply:
yup.  And empty enums should be accepted as a zero-width type.